src/realm/table.cpp

```diff
@@ -1289,17 +1396,21 @@ void Table::create_degen_subtab_columns()
     m_columns.update_parent();             // Throws
 
     Allocator& alloc = m_columns.get_alloc();
-    size_t num_cols = m_spec.get_column_count();
+    size_t num_cols = m_spec->get_column_count();
     for (size_t i = 0; i < num_cols; ++i) {
-        ColumnType type = m_spec.get_column_type(i);
-        bool nullable = (m_spec.get_column_attr(i) & col_attr_Nullable) != 0;
+        ColumnType type = m_spec->get_column_type(i);
+        int attr = m_spec->get_column_attr(i);
+        bool nullable = (attr & col_attr_Nullable) != 0;
+        // Must be 0, else there's no way to create search index for it statically
         size_t init_size = 0;
         ref_type ref = create_column(type, init_size, nullable, alloc); // Throws
         m_columns.add(int_fast64_t(ref));                               // Throws
 
-        // So far, only root tables can have search indexes, and this is not a
-        // root table.
-        REALM_ASSERT_3(m_spec.get_column_attr(i) & ~col_attr_Nullable, ==, col_attr_None);
+        // Create empty search index if required and add it to m_columns
+        if (attr & col_attr_Indexed) {
+            m_columns.add(StringIndex::create_empty(get_alloc()));
```

Comment:
Ah I now see where `create_empty()` is used. Is this an optimisation? Why not just add a full String Index that is attached to the column that was just created above?